swift-ticketing/src/swift_ticketing/db/booking.clj

```diff
@@ -1,25 +1,34 @@
 (ns swift-ticketing.db.booking
-  (:require [honey.sql :as sql]))
+  (:require [honey.sql :as sql]
+            [swift-ticketing.db.query :refer [run-query! run-query-one!]]))
 
 (defonce INPROCESS "InProcess")
 (defonce CONFIRMED "Confirmed")
 (defonce PAYMENTPENDING "PaymentPending")
 (defonce REJECTED "Rejected")
 (defonce CANCELED "Canceled")
 
-(defn insert-booking [uid booking-id]
-  (sql/format {:insert-into :booking
-                 :columns [:booking_id :user_id :booking_status]
-                 :values [[booking-id
-                           [:cast uid :uuid]
-                           [:cast INPROCESS :booking_status]]]}))
+(defn insert-booking [db-spec uid booking-id]
+  (run-query!
+   db-spec
+   (sql/format {:insert-into :booking
+                :columns [:booking_id :user_id :booking_status]
+                :values [[booking-id
+                          [:cast uid :uuid]
+                          [:cast INPROCESS :booking_status]]]})))
 
-(defn get-booking-status [uid booking-id]
-  (sql/format {:select [:booking_status] :from :booking
-               :where [[:= :booking_id [:cast booking-id :uuid]] 
-                       ]}))
+(defn get-booking [db-spec booking-id]
+  (run-query-one!
+   db-spec
+   (sql/format {:select [:*] :from :booking
+                :where [[:= :booking_id [:cast booking-id :uuid]]]})))
 
-(defn update-booking-status [booking-id booking-status]
-  (sql/format {:update :booking
-               :set {:booking_status [:cast booking-status :booking_status]}
-               :where [:= :booking_id [:cast booking-id :uuid]]}))
+(defn get-booking-status [db-spec booking-id]
+  (:booking_status (get-booking db-spec booking-id)))
+
+(defn update-booking-status [db-spec booking-id booking-status]
+  (run-query!
+   db-spec
```

Comment:
@neenaoffline what is the convention in clojure for db-spec? Could/Should it be a singleton instead of requiring app to pass it everytime?

Reply:
Discussed this on a call. Putting this down here.

We can use `component` to inject the DB component into the http component. But within the http component, I think we'll still need to pass it around explicitly.